app/experimenter/nimbus-ui/src/lib/mocks.tsx

```diff
@@ -60,3 +70,55 @@ export class MockedCache extends React.Component<MockedProps, MockedState> {
     this.state.client.stop();
   }
 }
+
+type ResultFunction = (operation: Operation) => FetchResult;
+
+type SimulatedMockedResponses = ReadonlyArray<{
+  request: MockedResponse["request"];
+  result: ResultFunction;
+  delay?: number;
+}>;
+
+export class SimulatedMockLink extends ApolloLink {
+  constructor(
+    public mockedResponses: SimulatedMockedResponses,
+    public addTypename: Boolean = true,
+  ) {
+    super();
+  }
+
+  public request(operation: Operation): Observable<FetchResult> | null {
+    const response = this.mockedResponses.find((response) =>
+      equal(operation.query, response.request.query),
+    );
+
+    if (!response) {
+      this.onError(
+        new Error(
+          `No more mocked responses for the query: ${print(
+            operation.query,
+          )}, variables: ${JSON.stringify(operation.variables)}`,
+        ),
+      );
+      return null;
+    }
+
+    const { result, delay } = response!;
+
+    return new Observable((observer) => {
+      let timer = setTimeout(
+        () => {
+          observer.next(
+            typeof result === "function"
+              ? (result(operation) as FetchResult)
+              : result,
+          );
+          observer.complete();
+        },
+        delay ? delay : 0,
+      );
+
+      return () => clearTimeout(timer);
+    });
+  }
+}
```

Comment:
I'm not actually sure what the rule is for the Nimbus app, but it seemed like at least for the rest of the codebase we're aiming for 100% coverage. These new mock utils would need to be covered, but also... they only exist for the purpose of testing, so 🤷

Reply:
Was tempted to find a different directory for this - it's mainly for stories, and we don't test stories (kind of circular & redundant). It could be used in tests, at which point it would be used. Not sure we can / should count test mocks as part of coverage (also kind of circular & redundant)